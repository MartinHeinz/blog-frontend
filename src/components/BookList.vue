<template>
    <section>
        <h1>Learning and Reading</h1>
        <p>These are books I read. My review and opinion about some of them can be found in blog posts.</p>
        <div class="books">
            <BookListItem v-for="item in items"
                          v-bind:src="item.src"
                          v-bind:alt="item.alt"
                          v-bind:url="item.url"
                          v-bind:key="item.id">
            </BookListItem>
        </div>
    </section>
</template>

<script>

import BookListItem from './BookListItem.vue';

export default {
    name: 'BookList',
    components: { BookListItem },
    props: {
        items: Array,
    },
};
</script>

<style scoped>
    section {
        display: block;
    }

    .books {
        text-align: center;
    }
</style>
