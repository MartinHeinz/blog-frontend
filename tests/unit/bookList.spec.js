import { createLocalVue, mount } from '@vue/test-utils';
import BookList from '@/components/BookList.vue';

const items = [
    {
        id: 1,
        created_at: '0001-01-01T00:00:00Z',
        updated_at: '0001-01-01T00:00:00Z',
        deleted_at: null,
        title: 'The Go Programming Language',
        src: 'https://www.gopl.io/cover.png',
        url: 'https://www.gopl.io/',
        alt: 'The Go Programming Language',
    },
    {
        id: 2,
        created_at: '0001-01-01T00:00:00Z',
        updated_at: '0001-01-01T00:00:00Z',
        deleted_at: null,
        title: 'Clean Code',
        src: 'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1436202607i/3735293._SX318_.jpg',
        url: 'https://www.goodreads.com/book/show/3735293-clean-code',
        alt: 'Clean Code: A Handbook of Agile Software Craftsmanship',
    },
];

describe('BookList.vue', () => {
    it('renders correcly', () => {
        const localVue = createLocalVue();
        const wrapper = mount(BookList, {
            localVue,
            propsData: {
                items,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
