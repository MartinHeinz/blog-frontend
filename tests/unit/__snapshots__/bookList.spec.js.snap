// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList.vue renders correcly 1`] = `
<section>
  <h1>Learning and Reading</h1>
  <p>These are books I read. My review and opinion about some of them can be found in blog posts.</p>
  <div class="books"><a href="https://www.gopl.io/" class="no-underline"><img src="https://www.gopl.io/cover.png" alt="The Go Programming Language" height="195" width="130"></a><a href="https://www.goodreads.com/book/show/3735293-clean-code" class="no-underline"><img src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1436202607i/3735293._SX318_.jpg" alt="Clean Code: A Handbook of Agile Software Craftsmanship" height="195" width="130"></a></div>
</section>
`;
